src/UserSpaceInstrumentation/Trampoline.cpp

```diff
@@ -77,6 +77,60 @@ constexpr uint64_t kOffsetOfFunctionIdInCallToEntryPayload = 178;
   return result;
 }
 
+// Check if somewhere in the code of `function` there is a (conditional) jump back to the first five
+// bytes of the function (which we intend to overwrite with a jump into the trampoline). If so we
+// must not instrument the function. Note that the entire function is not necessarily available
+// here; we'll just disassemble and check whatever we have.
+// This is merely a heuristic. There can be other jumps either further down in the function or in
+// different places in the same translation unit that target the first five bytes of a function.
+// However analysing existing code shows that many of the problematic jumps are in small functions
+// that are written in assembly. These are detected by the function below.
+bool CheckForJumpIntoFirstFiveBytes(uint64_t function_address, const std::vector<uint8_t>& function,
```

Comment:
Since the checks are only for relative jumps, maybe rename to "CheckForRelativeJumpIntoFirstFiveBytes"?